docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc

```diff
@@ -0,0 +1,96 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="Concepts for sizing CPU and memory resources"
+summary="Understand these concepts to avoid resource exhaustion and congestion"
+preview="true"
+tileVisible="false" >
+
+Use this as a starting point to size a product environment. 
+Adjust the values for your environment as needed based on your load tests.
+
+== Performance recommendations
+
+[WARNING]
+====
+* Performance will be lowered when scaling to more Pods (due to additional overhead) and using a cross-datacenter setup (due to additional traffic and operations).
+
+* Increased cache sizes can improve the performance when {project_name} instances run for a longer time. Still, those caches need to be filled when an instance is restarted.
+
+* Use these values as a starting point and perform your own load tests before going into production.
+====
+
+Summary:
+
+* The used CPU scales linearly with the number of requests up to the tested limit below.
+* The used memory scales linearly with the number of active sessions up to the tested limit below.
+
+Recommendations:
+
+* The base memory usage for an inactive Pod is 1 GB of RAM.
+
+* Leave 1 GB extra head-room for spikes of RAM.
+
+* For each 100,000 active user sessions, add 500 MB per Pod in a three-node cluster (tested with up to 200,000 sessions).
++
+This assumes that each user connects to only one client.
+Memory requirements increase with the number of client sessions per user session (not tested yet).
+
+* For each 40 user logins per second, 1 vCPU per Pod in a three-node cluster (tested with up to 300 per second).
++
+{project_name} spends most of the CPU time hashing the password provided by the user.
+
+* For each 450 client credential grants per second, 1 vCPU per Pod in a three node cluster (tested with up to 2000 per second).
```

Comment:
Not sure if deserves a note about this but that is probably based on a client setting where refresh tokens are disabled (default). Some people might be using or might want (don't ask me why but sadly we used to support that) refresh tokens when doing client credentials and that might change the results a lot.

Reply:
Will add a sentence to the reference architecture in https://github.com/keycloak/keycloak/issues/24975